pkg/services/publicdashboards/service/service.go

```diff
@@ -224,3 +237,25 @@ func GenerateAccessToken() (string, error) {
 
 	return fmt.Sprintf("%x", token[:]), nil
 }
+
+func (pd *PublicDashboardServiceImpl) getSafeIntervalAndMaxDataPoints(reqDTO *PublicDashboardQueryDTO, ts *TimeSettings) (int64, int64) {
```

Comment:
I think this makes sense. Could we add documentation here that describes what we deem safe to be?